packages/vite/src/node/plugins/resolve.ts

```diff
@@ -899,8 +906,23 @@ export async function tryOptimizedResolve(
     // lazily initialize resolvedSrc
     if (resolvedSrc == null) {
       try {
-        // this may throw errors if unable to resolve, e.g. aliased id
-        resolvedSrc = normalizePath(resolveFrom(id, path.dirname(importer)))
```

Comment:
`resolveFrom` resolve to `package.json#main` field, not respecting vite's main fields (`module` and custom mainFields).